diff --git a/Sources/Grammar/GrammarAnalysis/LeftFactoring.swift b/Sources/Grammar/GrammarAnalysis/LeftFactoring.swift
index aa37488..32bf6d0 100644
--- a/Sources/Grammar/GrammarAnalysis/LeftFactoring.swift
+++ b/Sources/Grammar/GrammarAnalysis/LeftFactoring.swift
@@ -7,6 +7,8 @@
 //
 
 import Foundation
+#if canImport(OSLog)
 import OSLog
+#endif
 
 extension Grammar {
diff --git a/Sources/Grammar/GrammarAnalysis/LeftRecursion.swift b/Sources/Grammar/GrammarAnalysis/LeftRecursion.swift
index 256f964..f62a124 100644
--- a/Sources/Grammar/GrammarAnalysis/LeftRecursion.swift
+++ b/Sources/Grammar/GrammarAnalysis/LeftRecursion.swift
@@ -7,6 +7,8 @@
 //
 
 import Foundation
+#if canImport(OSLog)
 import OSLog
+#endif
 
 /// Eliminating Left Recursion is described in detail in the following books:
diff --git a/Sources/Grammar/GrammarAnalysis/Nullable.swift b/Sources/Grammar/GrammarAnalysis/Nullable.swift
index e3caf0e..d573d60 100644
--- a/Sources/Grammar/GrammarAnalysis/Nullable.swift
+++ b/Sources/Grammar/GrammarAnalysis/Nullable.swift
@@ -7,6 +7,8 @@
 //
 
 import Foundation
+#if canImport(OSLog)
 import OSLog
+#endif
 
 extension Grammar {
diff --git a/Sources/Grammar/GrammarLogger.swift b/Sources/Grammar/GrammarLogger.swift
index 3329324..b828a1d 100644
--- a/Sources/Grammar/GrammarLogger.swift
+++ b/Sources/Grammar/GrammarLogger.swift
@@ -6,6 +6,14 @@
 //  Copyright © 2025 hakkabon software. All rights reserved.
 //
 
+#if canImport(OSLog)
 import OSLog
+#else
+struct Logger {
+    init(subsystem: String, category: String) {}
+    func info(_ message: @autoclosure () -> String) {}
+    func trace(_ message: @autoclosure () -> String) {}
+}
+#endif
 
 extension Logger {
diff --git a/Sources/Grammar/GrammarNotation/StandardForm.swift b/Sources/Grammar/GrammarNotation/StandardForm.swift
index fc4d1ca..3563d0d 100644
--- a/Sources/Grammar/GrammarNotation/StandardForm.swift
+++ b/Sources/Grammar/GrammarNotation/StandardForm.swift
@@ -7,6 +7,8 @@
 //
 
 import Foundation
+#if canImport(OSLog)
 import OSLog
+#endif
 
 /// Transforms an extended BNF grammar into Standard notation (BNF).
diff --git a/Sources/Grammar/GrammarParser/GrammarParser.swift b/Sources/Grammar/GrammarParser/GrammarParser.swift
index ab95af9..7d2f6a1 100644
--- a/Sources/Grammar/GrammarParser/GrammarParser.swift
+++ b/Sources/Grammar/GrammarParser/GrammarParser.swift
@@ -7,7 +7,9 @@
 //
 
 import Foundation
+#if canImport(OSLog)
 import OSLog
+#endif
 import Tokenizer
 
 /// A slightly modified version of WSN (Wirth Syntax Notation) to accomodate
